README.md

```diff
@@ -355,6 +356,75 @@ writable.
 
 Always prefer `ReadonlyArray` over a regular `Array` unless it must be possible to modify the Array.
 
+### React props & TypeScript
+
+#### Why use Interface instead of Type for typing props?
+
+`interface` are a little more powerful than the corresponding `type` declaration.
+
+One of the reasons to use `interface` is when you need to compose two or more types, you have the
+option of extending those types with an interface, or intersecting them in a type alias. TS team
+says:
+
+> Interfaces create a single flat object type that detects property conflicts, which are usually
+> important to resolve! Intersections on the other hand just recursively merge properties, and in
+> some cases produce never. Interfaces also display consistently better, whereas type aliases to
+> intersections can't be displayed in part of other intersections. Type relationships between
+> interfaces are also cached, as opposed to intersection types as a whole. A final noteworthy
+> difference is that when checking against a target intersection type, every constituent is checked
+> before checking against the "effective"/"flattened" type. For this reason, extending types with
+> interfaces/extends is suggested over creating intersection types.
+
+This is preferred:
+
+```ts
+interface Foo extends Bar, Baz {
+  someProp: string;
+}
+```
+
+Instead of this:
+
+```ts
+type Foo = Bar &
+  Baz & {
+    someProp: string;
+  };
+```
+
+#### Why not use React.FC or FunctionComponent type?
+
+- Provides an implicit definition of `children`
+- Doesn't support generics.
+- Doesn't work correctly with `defaultProps`
+
+#### Implicit or explicit `returnType`?
+
+It depends. If we know we want to return a JSX element, it is not necessary to type the `returnType`
+because it is inferred from what you actually return, so can be omitted most of the time.
+
+#### Preferred approach
+
+The best approach and example would be something like this:
+
+```ts
+interface FancyButtonProps {
+  color: string;
+}
+
+export default function FancyButton({ color }: FancyButtonProps) => {
+  //...
+};
+```
+
+or if we want to explicity type the `returnType`:
+
+```ts
+export default function FancyButton({ color }: FancyButtonProps): JSX.Element => {
```

Comment:
I have no idea how we missed this, but this isn't a valid function definition. The `=>` after `JSX.Element` is only used for well arrow functions and here you managed to combine both a standard function and an arrow function.

See [https://www.typescriptlang.org/play](https://www.typescriptlang.org/play?#code/FASwdgLgpgTgZgQwMZQAQDEFiQTwEICuEEA9mAAowkAOAzqgN7CoupIkA2JMAXKrRBjgA5gG5gAX2DSA9DP4QsAEwQwlqOAWwQQZYJu26wGLLkLEyACgZtO3VBL6Zs+IqQpU6ASj5aA1mAkAO7GTJKy8qpUQRpaSDp67GACJi7m7qgAvKjWtlwwDk6mrhYeNLQ+qP6BIVkAfIzSUsByqBAAFmgCyqrqyqhRwbGGZKhcAG5QSO0gHEoAhPpxCcbOZm5WNuz5hanrpZTlldXBxpkNYRJAA)

Reply:
Good spot

Reply:
@psimk @ThijsTyZ dummy mistake!

Already fixed.